client/pages/index.tsx

```diff
@@ -1,37 +1,50 @@
-import { Link, withAuth } from '@components';
+import { useState, useEffect } from 'react';
+import { Formik, Form } from 'formik';
+import { Link, withAuth, Stepper, Button } from '@components';
 import type { NextPage } from 'next';
 import dynamic from 'next/dynamic';
+import { PlanningSteps } from '../constants';
+import { FormContent } from '../components/forms';
+import { PageTitle } from 'components/PageTitle';
 
 const Dashboard: NextPage = () => {
+  const [currentStep, setCurrentStep] = useState(1);
+  const isFirstStep = currentStep === 1;
+
+  const handleClick = (clickType: any) => {
+    let newStep = currentStep;
+    clickType == 'next' ? newStep++ : newStep--;
```

Comment:
Rather than a parameter here, this would be better as two separate functions ie: `handleNext/handlePrevious`

The inconsistent calling of this method in `onClick={handleClick}` and `onClick={() => handleClick('next')}` is a smell to me 

Having separate functions with a single purpose would be cleaner 

```
handleNext()
  if(currentStep <= PlanningSteps.length)
    setCurrentStep(step => step + 1)
```